porter/v1/porter_app.proto

```diff
@@ -152,3 +152,39 @@ message EFS {
   // file_system_id is the AWS EFS File System ID to mount
   string file_system_id = 2;
 }
+
+// EnvDefinitionMethod is the method for setting the environment variable
+enum EnvDefinitionMethod {
+  ENV_DEFINITION_METHOD_UNSPECIFIED = 0;
+  ENV_DEFINITION_METHOD_FROM_VALUE = 1;
+  ENV_DEFINITION_METHOD_FROM_APP = 2;
+}
+
+message EnvVariable {
+  // key is the name of the environment variable
+  string key = 1;
+  // definition_method specifies how the env variable should be set
+  EnvDefinitionMethod definition_method = 2;
```

Comment:
Is this necessary? Could we just use orders of preference .i.e cli flag override value overrides from

Reply:
I used this to switch on different cases but I could replace that with nil checks in code if that makes the contract clearer